lightning/src/ln/channel.rs

```diff
@@ -301,6 +301,10 @@ enum ChannelState {
 	/// We've successfully negotiated a closing_signed dance. At this point ChannelManager is about
 	/// to drop us, but we store this anyway.
 	ShutdownComplete = 4096,
+	/// Flag which is set on `FundingSent` to indicate this channel is funded in a batch and the
+	/// broadcasting of the funding transaction is being held until all channels in the batch
+	/// have received funding_signed and have their monitors persisted.
+	WaitingForBatch = 1 << 13,
```

Comment:
Makes sense! I think this will depend on whether `Channel`s with `WaitingForBatch` will be serialized. If we don’t, that would would sidestep this issue. Discussed this in a comment below.

Reply:
As discussed below, `Channel`s with `WaitingForBatch` won't be serialized. Would it be necessary to still ensure this for safety in this PR?